### ISSUE TITLE
Intel MKL ERROR: Parameter 6 was incorrect on entry to DGELSD.

### ISSUE THREAD

**Issue by maoshi7** [bubbliiiing/unet-pytorch#118]
数月以前，我使用这里提供的接近原版U-Net的网络，在某数据集上重新训练时，服务器终端中每一epoch都提示Intel MKL ERROR: Parameter 6 was incorrect on entry to DGELSD.且loss都为nan，而我在我自己的电脑上使用同样的程序进行训练就没错。我百思不得其解。后来，我将服务器整个工程文件夹删除，从本人电脑重新上传整个工程文件夹，消除了这个报错。但最近，我又遇到了这个问题。这次不是从一开始就报这个错，而是训练到第47epoch时，训练过程中loss突然变成nan了，然后就报Intel MKL ERROR: Parameter 6 was incorrect on entry to DGELSD.，再之后都是在报错。而这次训练的模型，之前也训练过了，我这次只是想重新训练，检查它的可靠性，没想到会出这个问题。为什么多次出现同一套程序一开始运行得没问题，反复多运行几次就开始报这个错误的情况？是程序之中自有隐性bug吗？有没有人遇到过和我一样的困境？总不能再删除服务器所有文件，然后重新上传吧？且不说这次还能不能有用，这也本就不是治本的办法。
还有一个困境就是有时就算不报这个错，loss也一直卡在0.5不下降了，网络学习不到任何特征，识别不了任何一个目标。

**Comment by maoshi7** [comment]
English Version:
Several months ago, when retraining a near-original U-Net network (provided here) on a dataset, the server terminal repeatedly showed:
Intel MKL ERROR: Parameter 6 was incorrect on entry to DGELSD.
with loss values becoming NaN every epoch. The same program ran flawlessly on my local machine. After extensive troubleshooting, I resolved it by deleting the entire project folder on the server and re-uploading everything from my local computer.

Recently, the issue resurfaced:

Training proceeded normally until epoch 47, when loss suddenly turned NaN
Followed by the same Intel MKL ERROR: Parameter 6 was incorrect on entry to DGELSD.
Subsequent epochs consistently failed
This model had trained successfully before – this retraining was for reliability verification
Core concerns:

Why does the same code fail intermittently after initial successful runs?
Could there be hidden bugs in the implementation?
Has anyone encountered similar issues?
Re-uploading all files isn't a sustainable solution
Additional challenge:
Even without this error, loss sometimes stagnates at 0.5 – the network fails to learn features or detect any targets.

**Comment by maoshi7** [comment]
天呐，我重新运行，竟然又让我遇到新情况了！之前是total loss和val loss从某个epoch开始都为nan,的轮次就都是nan了；这次遇到了某一轮次total loss和val loss变为nan，但下一轮次total loss能正常计算了，val loss仍然是nan，之后的轮次就都是这样了。Total loss还在缓慢下降，说明梯度反向传播都没出错，但怎么会无法在验证集上进行运行了呢？！

![Image](https://github.com/user-attachments/assets/7c185c16-1111-478e-acca-3bed3b547a05)

**Comment by nanfangxiansheng** [comment]
same question
